verification-right.md

```diff
@@ -0,0 +1,139 @@
+# ✅ Verification: the right way
+
+Did you notice anything wrong with our verification in the previous step?
+
+Nothing? Let's see:
+
+* You've checked that the `attestedClaim` data is valid and you've verified that the corresponding attestation is on chain and not revoked. All good.
+
+* But: are you sure that the entity/person that sent you the `attestedClaim` owns it?
+What if a malicious actor stole this `attestedClaim` and is now presenting it to you as theirs? We'll see how to mitigate this.
+
+## Understand credential theft mitigation
+
+To mitigate credential theft, a <span class="label-role verifier">verifier</span> can initiate a **cryptographic challenge** with a <span class="label-role claimer">claimer</span>.
+
+The underlying idea is simple: to prove their identity, the <span class="label-role claimer">claimer</span> signs **on-the-fly** - that's important - a piece of data under the same identity as the identity associated with their `attestedClaim`. By checking this signature's validity, the <span class="label-role verifier">verifier</span> makes sure that the `attestedClaim` is owned by the person who just sent it.
+
+What piece of data should be signed? It doesn't really matter; it can be an arbitrary number picked by the <span class="label-role verifier">verifier</span>. What matters is that this number should be used only once. Otherwise, the cryptographic challenge is worthless.
+
+> 💡 In a cryptographic communication, an arbitrary number that can be used just once is called a **nonce**.
+
+Here's how it works:
+
+1. The <span class="label-role verifier">verifier</span> sends a nonce to the <span class="label-role claimer">claimer</span>.
+2. The <span class="label-role claimer">claimer</span> sends back this nonce signed with their **private** key, together with their `attestedClaim`.
+3. The <span class="label-role verifier">verifier</span> checks the following:
+   * Does the signature on the nonce match the public key contained in the `attestedClaim`? If so: the entity/person who just sent the `attestedClaim` plus the signed nonce is also the owner of this `attestedClaim`. If not: the `attestedClaim` might be stolen.
+   * Is the data valid? Is the attestation on-chain and not revoked? See the simple [Verification](verification).
+
+OK, let's see this in action.
+
+## As the <span class="label-role verifier">verifier</span>: create a challenge
+
+To generate a random, unique piece of data, we'll use a package called [uuid].
+Install it:
+
+```bash
+# with yarn
+yarn add uuid
+# or with npm
+npm install uuid
+```
+
+Create a new file `challenge.js`, and paste the following code into it:
+
+```javascript
+import { v4 as uuid } from 'uuid'
```

Comment:
This should be a `require` and no `import`?

Reply:
👍 right, fixed